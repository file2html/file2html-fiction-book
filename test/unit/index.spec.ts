import * as fs from 'fs';
import * as path from 'path';
import FictionBookReader from '../../src/index';

describe('FictionBook', () => {
    describe('#read()', () => {
        it('should parse a .fb2 sample', () => {
            const filename: string = path.resolve(__dirname, '..', 'sample.fb2');
            const fileBuffer: Buffer = fs.readFileSync(filename);
            const charCodes: number[] = fs.readFileSync(filename, {
                encoding: 'utf8'
            }).split('').map((char: string) => char.charCodeAt(0));

            return new FictionBookReader().read({
                fileInfo: {
                    content: Object.assign(charCodes, {
                        byteLength: fileBuffer.byteLength
                    }),
                    meta: {} as any
                }
            }).then((file) => {
                expect(file.getMeta()).toEqual({
                    createdAt: '2002-10-15',
                    creator: 'GribUser',
                    fileType: 1,
                    mimeType: '',
                    modifiedAt: '',
                    name: '',
                    size: 37837
                });
                expect(file.getData().styles).toBe('<style></style>');
                expect(file.getData().content).toBe('<div></div>');
            });
        });
    });
});