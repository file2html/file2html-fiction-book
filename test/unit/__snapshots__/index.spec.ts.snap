// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles 1`] = `
"
            <style>
                .subtitle {text-align: center}
            </style>
        "
`;
